src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+        num_samps (int): Number of samples per excerpt for the filter bank corresponding to 2^16 samples at sf=250.
+        segment_boundaries (list): First and last indices analyzed within the ECG signal.
+        signal_segment (np.ndarray): The ECG signal designated for analysis.
+        messages (dict): Dictionary containing information relevant to the algorithm.
+
+    Returns:
+        signal_segment (np.ndarray): The ECG signal designated for analysis.
+        current_samp (int): Updated sample analyzed within the ECG signal.
+        updated_samp (int): Updated current sample analyzed within the ECG signal.
+        synchronized_wavelet_matrix (np.ndarray): Wavelet Transform Matrix, relevant to the current segment of the signal.
+        threshold_matrix (np.ndarray): QRS detection threshold matrix.
+        end_samp (int): Updated final sample analyzed within the ECG signal.
+        initial_samp (int): Updated initial sample analyzed within the ECG signal.
+
+    NOTE:
+        First l5-1 samples are not correctly filtered (border effect)
+        Last d5 samples are discarded in order to alineate all the
+        filtered signals, taking into acount the filter delays.
+    """
+
+    def process_filters(quadratic_filter_dictionary):
+        required_keys = {"filters", "lengths", "d_values"}
+        keys_in_dict = set(quadratic_filter_dictionary.keys())
+        missing_keys = required_keys - keys_in_dict
+        extra_keys = keys_in_dict - required_keys
+        if missing_keys:
+            raise ValueError(
+                f"quadratic_filter_dictionary is missing keys: {missing_keys}"
+            )
+        if extra_keys:
+            raise ValueError(
+                f"quadratic_filter_dictionary contains unexpected keys: {extra_keys}"
+            )
+        quadratic_filter_array, lengths, decimation_values = [
+            quadratic_filter_dictionary[key] for key in required_keys
+        ]
+        if not quadratic_filter_array or not lengths or not decimation_values:
+            raise ValueError(
+                "quadratic_filter_dictionary contains empty values for 'filters', 'lengths', or 'd_values'."
+            )
+        return
+
+    # Validation
+    process_filters(quadratic_filter_dictionary)
+    for component in (current_samp, initial_samp, num_samps):
+        if (not isinstance(component, int)) or (component < 0):
+            raise ValueError(f"{component} must be a non-negative integer.")
+
+    if signal_segment.size == 0:
+        raise ValueError("Input is an empty array.")
+    if signal_segment.squeeze().ndim > 1:
+        raise ValueError("Input is a higher-dimensional array, not a matrix.")
+
+    if not (
+        isinstance(segment_boundaries, list)
+        and all(isinstance(x, int) for x in segment_boundaries)
+        and len(segment_boundaries) == 2
+    ):
+        raise ValueError(
+            "segment_boundaries must be a non-empty list of integers or floats."
+        )
```

Comment:
IMO, two variables (e.g. `start, end`) with type annotations is clearer then a list like:
```python
segment_boundaries = [start, end]
```